src/cdk-experimental/menu/menu-tree-service.ts

```diff
@@ -0,0 +1,26 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Track the open menu HTML elements in order to handle click events outside the menu tree when the
+ * menus aren't logically nested.
+ */
+export class OpenMenuTracker {
+  /** Array of the HTMLElements in the open menu tree starting from the root Menu Bar. */
+  readonly openMenus: HTMLElement[] = [];
```

Comment:
I *might* have you use the Menu framework type here, instead of the raw, internal HTMLElement. That could make this class a little more robust to implementation changes. You could have a menu to get the HTMLElements for the purposes of configuring the overlay.

@jelbourn, does that seem like a reasonable idea?

Reply:
I would rather use the native elements here to avoid an extra level of indirection. As long as this remains an internal API, I don't think it matters much.